engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -868,24 +875,31 @@ public long nextSequenceId() {
   }
 
   public ExecutionEnvironment getExecutionEnvironment() {
-    return executionEnvironment;
+    return contextExecutionEnvironment;
+  }
+
+  @TruffleBoundary
```

Comment:
I don't think we want `@TruffleBoundary` here. Quite opposite. We want this to be fast!

Reply:
Can you remove the boundary, @4e6, or are there some problems that prevent it to be removed?